redis/tests/test_types.rs

```diff
@@ -19,324 +19,378 @@ fn test_is_single_arg() {
     assert!(!twobytesvec.is_single_arg());
 }
 
-#[test]
-fn test_info_dict() {
-    use redis::{FromRedisValue, InfoDict, Value};
+/// The `FromRedisValue` trait provides two methods for parsing:
+/// - `fn from_redis_value(&Value) -> Result<T, RedisError>`
+/// - `fn from_owned_redis_value(Value) -> Result<T, RedisError>`
+/// The `RedisParseMode` below allows choosing between the two
+/// so that test logic does not need to be duplicated for each.
+enum RedisParseMode {
+    Owned,
+    Ref,
+}
 
-    let d: InfoDict = FromRedisValue::from_redis_value(&Value::Status(
-        "# this is a comment\nkey1:foo\nkey2:42\n".into(),
-    ))
-    .unwrap();
+impl RedisParseMode {
+    /// Calls either `FromRedisValue::from_owned_redis_value` or
+    /// `FromRedisValue::from_redis_value`.
+    fn parse_redis_value<T: redis::FromRedisValue>(
+        &self,
+        value: &redis::Value,
```

Comment:
Possibly; this way just required fewer changes since I didn't need to audit all of the existing calls in the test, they could just be mechanically replaced.

Reply:
I just made this change